sample-code/vc-using-framestream/vc-using-framestream/Connection.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+#include "commands.h"
+#include "AcousticSettings.h"
+#include "FrameStreamListener.h"
+#include "UdpListener.h"
+#include <memory>
+#include <boost/asio.hpp>
+
+namespace aris { class Command; }
+
+class Connection
+{
+public:
+  // Creates the Connection object with a live connection to the ARIS.
+  // Returns an unassociated unique_ptr if the connection could not be made.
+  static std::unique_ptr<Connection> Create(
+    boost::asio::io_service & io, 
+    const boost::asio::ip::tcp::endpoint & ep,
+    std::function<void(Aris::Network::FrameBuilder&)> onFrameCompletion,
+    SystemType systemType,
+    aris::Command::SetSalinity::Salinity salinity,
+    float initialFocusRange,
```

Comment:
The focus range is so dependent on the acoustic settings (window start and end) that I think it might make more sense to just derive it for them rather than have them pass it in.